lib/http/uri_normalizer.rb

```diff
@@ -0,0 +1,25 @@
+require "uri"
+
+module HTTP
+  module URINormalizer
+    module_function
+
+    # Normalize URI
+    #
+    # @param [#to_s] uri
+    # @return [::URI]
+    def normalize_uri(uri)
```

Comment:
Well, I'm 100% agree that this hack is ugly as <put any word you think more appropriate>.
And I'm OK with Addressable, but not as hard dependency.